env/HERA.env

```
@@ -157,167 +140,128 @@ elif [[ "${step}" = "anal" ]] || [[ "${step}" = "analcalc" ]]; then
     export APRUN_CALCINC="${launcher} \$ncmd --cpus-per-task=${NTHREADS_CALCINC}"
 
-    export NTHREADS_CYCLE=${nth_cycle:-12}
-    [[ ${NTHREADS_CYCLE} -gt ${npe_node_max} ]] && export NTHREADS_CYCLE=${npe_node_max}
-    npe_cycle=${ntiles:-6}
-    export APRUN_CYCLE="${launcher} -n ${npe_cycle} --cpus-per-task=${NTHREADS_CYCLE}"
+    export NTHREADS_CYCLE=${threads_per_task_cycle:-12}
+    [[ ${NTHREADS_CYCLE} -gt ${max_tasks_per_node} ]] && export NTHREADS_CYCLE=${max_tasks_per_node}
+    ntasks_cycle=${ntiles:-6}
+    export APRUN_CYCLE="${launcher} -n ${ntasks_cycle} --cpus-per-task=${NTHREADS_CYCLE}"
 
     export NTHREADS_GAUSFCANL=1
-    npe_gausfcanl=${npe_gausfcanl:-1}
-    export APRUN_GAUSFCANL="${launcher} -n ${npe_gausfcanl} --cpus-per-task=${NTHREADS_GAUSFCANL}"
+    ntasks_gausfcanl=${ntasks_gausfcanl:-1}
+    export APRUN_GAUSFCANL="${launcher} -n ${ntasks_gausfcanl} --cpus-per-task=${NTHREADS_GAUSFCANL}"
 
 elif [[ "${step}" = "sfcanl" ]]; then
 
-    nth_max=$((npe_node_max / npe_node_sfcanl))
-
-    export NTHREADS_CYCLE=${nth_sfcanl:-14}
-    [[ ${NTHREADS_CYCLE} -gt ${npe_node_max} ]] && export NTHREADS_CYCLE=${npe_node_max}
-    npe_sfcanl=${ntiles:-6}
-    export APRUN_CYCLE="${launcher} -n ${npe_sfcanl} --cpus-per-task=${NTHREADS_CYCLE}"
+    export NTHREADS_CYCLE=${threads_per_task:-14}
+    export APRUN_CYCLE="${APRUN} --cpus-per-task=${NTHREADS_CYCLE}"
 
 elif [[ "${step}" = "eobs" ]]; then
 
     export MKL_NUM_THREADS=4
     export MKL_CBWR=AUTO
 
-    nth_max=$((npe_node_max / npe_node_eobs))
-
-    export NTHREADS_GSI=${nth_eobs:-${nth_max}}
-    [[ ${NTHREADS_GSI} -gt ${nth_max} ]] && export NTHREADS_GSI=${nth_max}
-    export APRUN_GSI="${launcher} -n ${npe_gsi:-${npe_eobs}} --cpus-per-task=${NTHREADS_GSI}"
+    export NTHREADS_GSI=${NTHREADSmax}
+    export APRUN_GSI="${APRUN} --cpus-per-task=${NTHREADS_GSI}"
 
     export CFP_MP=${CFP_MP:-"YES"}
     export USE_CFP=${USE_CFP:-"YES"}
     export APRUNCFP="${launcher} -n \$ncmd ${mpmd_opt}"
 
 elif [[ "${step}" = "eupd" ]]; then
 
-    nth_max=$((npe_node_max / npe_node_eupd))
-
-    export NTHREADS_ENKF=${nth_eupd:-${nth_max}}
-    [[ ${NTHREADS_ENKF} -gt ${nth_max} ]] && export NTHREADS_ENKF=${nth_max}
-    export APRUN_ENKF="${launcher} -n ${npe_enkf:-${npe_eupd}} --cpus-per-task=${NTHREADS_ENKF}"
+    export NTHREADS_ENKF=${NTHREADSmax}
+    export APRUN_ENKF="${launcher} -n ${ntasks_enkf:-${ntasks}} --cpus-per-task=${NTHREADS_ENKF}"
 
     export CFP_MP=${CFP_MP:-"YES"}
     export USE_CFP=${USE_CFP:-"YES"}
     export APRUNCFP="${launcher} -n \$ncmd ${mpmd_opt}"
 
 elif [[ "${step}" = "fcst" ]] || [[ "${step}" = "efcs" ]]; then
 
-    ppn="npe_node_${step}_${RUN}"
-    [[ -z "${!ppn+0}" ]] && ppn="npe_node_${step}"
-    nprocs="npe_${step}_${RUN}"
-    [[ -z ${!nprocs+0} ]] && nprocs="npe_${step}"
+    ppn="tasks_per_node"
+    [[ -z "${!ppn+0}" ]] && ppn="tasks_per_node"
+    nprocs="ntasks"
+    [[ -z ${!nprocs+0} ]] && nprocs="ntasks"
```

Setting the variable again to the same thing isn't going to fix an issue.
```suggestion
    ppn="tasks_per_node"
    nprocs="ntasks"
```

---

Also, why is the formulation for Hera and WCOSS different from other machines?

---

`ppn` and `nprocs` are not needed anymore.  I will delete these from `HERA.env` and `WCOSS2.env`.

---

Fixed.